rules/dead-doc-block/tests/Rector/Node/RemoveNonExistingVarAnnotationRector/Fixture/skip_other_comment_before_var.php.inc

```diff
@@ -0,0 +1,15 @@
+<?php
+
+namespace Rector\DeadDocBlock\Tests\Rector\Node\RemoveNonExistingVarAnnotationRector\Fixture;
+
+class SkipOtherCommentBeforeVar
+{
+    public function get()
+    {
+        /** @var \stdClass $nonExisting */
+        // Load data also with projekt...
+        $return[] = $this->getReturnData();
```

Comment:
The `@var` annotation should be removed here, as it relates to non-existing variable

```suggestion
        // Load data also with projekt...
        $return[] = $this->getReturnData();
```

Reply:
I tried that, it somehow remove all comments together:

```php
        $comments = $node->getComments();
        if (isset($comments[1]) && $comments[1] instanceof Comment) {
            $node->setAttribute(AttributeKey::PHP_DOC_INFO, $phpDocInfo);
            $node->setAttribute(AttributeKey::COMMENTS, [$comments[1]]);

            return $node;
        }
```

Reply:
I see.

It might be related to adding docblocks back to nodes in BetterStandardPrinter:
https://github.com/rectorphp/rector/blob/a4f8a23ecf5019f9344159707890c90722a0ae52/src/PhpParser/Printer/BetterStandardPrinter.php#L585

Reply:
I tried add: 

```php
$node->setAttribute(AttributeKey::COMMENTS, null);
```
in 

https://github.com/rectorphp/rector/blob/d4883d44d925b80ce4293f8f71dada290baf06ef/packages/node-type-resolver/src/PhpDoc/NodeAnalyzer/DocBlockManipulator.php#L140

That remove all comments, but I can't re-add the comment back in the rule

Reply:
finally fixed.